tasks/download_file.sh

```diff
@@ -1,54 +1,17 @@
-#!/usr/bin/env bash
-# shellcheck disable=SC2154
+#!/bin/bash
+# shellcheck disable=SC2154,SC1090,SC2128
 
-URL=$PT_url
-DESTINATION=$PT_destination
-DESTINATION_DIR=$(dirname "$DESTINATION")
+declare PT__installdir
+source "$PT__installdir/deploy_pe/files/common.sh"
 
-function raise_error {
-  cat << ERROR_MESSAGE
-  { "_error": {
-    "msg": "Task exited 1:\\n $1",
-    "kind": "deploy_pe/download-error",
-    "details": { "exitcode": 1 }
-    }
-  }
-ERROR_MESSAGE
-  exit 1
-}
+(( EUID == 0 )) || fail "This utility must be run as root"
 
-function output {
-  cat << OUTPUT_MESSAGE
-  { 
-    "file": "$1",
-    "msg": "$2"
-  }
-OUTPUT_MESSAGE
-  exit 0
-}
+_tmp_dir="$(mktemp -d)"
```

Comment:
Under this logic, we would download a new copy of the installer into a temp directory for every time this task was run. We may want to clean up the files or think about moving the installer to an accessible location if needed to ensure we don't fill up the partition if the plan is run multiple times.

Reply:
I've considered having something in `common.sh` to remove temp files with a certain prefix, but I'm not sure whether it's a good idea or not yet.  We could have a special case in this script that cleans up the installer, though.

Reply:
I end up using the installer sometimes in the debugging kit. Maybe we can have an option to move it to `$HOME` at the end so we would at maximum have 2 instances? I guess we could also just run it from there as well, but that seems less flexible.

Reply:
Is there a large downside to just leaving it in the temp dir?  `/tmp` is cleaned up every 24 hours.  It's also trivial for us to just download the installer if you're manually doing something.